### ISSUE TITLE
serve: receiving unexpected EOF responses

### ISSUE THREAD

**Issue by shayne** [tailscale/tailscale#6958]
A user reported that their Funnel/serve stops working after a long period of time (or, possibly, when resuming from sleep). The only log line I've found that provides any indication of what's going on is from [ipnlocal/serve.go](https://github.com/tailscale/tailscale/blob/82b9689e25d1a1a0dac541381057013ced6a9ff9/ipn/ipnlocal/serve.go#L236-L240):
```
localbackend: got ingress conn for unconfigured %q; rejecting
```

The user who reported the bug says running `tailscale down` followed by `tailscale up` resolves the issue. My running theory is `LocalBackend` resetting its state (clearing `serveConfig`), caused by sleep/loss of network. 

Internal discussion: https://tailscale.slack.com/archives/CTU3D5HAS/p1673628165634569
Support ticket: https://secure.helpscout.net/conversation/2122907975/8572

<details>
<summary><samp>ipnlocal/local.go:2122</samp> <code>func (b *LocalBackend) setAtomicValuesFromPrefsLocked(p ipn.PrefsView)</code></summary>

https://github.com/tailscale/tailscale/blob/61dfbc0a6e91be32580553d15d8b12a4c955a6bc/ipn/ipnlocal/local.go#L2120-L2134
</details>

<details>
<summary><samp>ipnlocal/local.go:3766</samp> <code>func (b *LocalBackend) reloadServeConfigLocked(prefs ipn.PrefsView)</code></summary>

https://github.com/tailscale/tailscale/blob/61dfbc0a6e91be32580553d15d8b12a4c955a6bc/ipn/ipnlocal/local.go#L3766-L3794
</details>

<details>
<summary><samp>ipnlocal/local.go:4572</samp> <code>func (b *LocalBackend) resetForProfileChangeLockedOnEntry() error</code></summary>

https://github.com/tailscale/tailscale/blob/61dfbc0a6e91be32580553d15d8b12a4c955a6bc/ipn/ipnlocal/local.go#L4571-L4584
</details>

The next step would be to reproduce the bug and test the theory of resuming from sleep/loss of network.

**Comment by mvdan** [comment]
:wave: happy to provide more details if needed. I haven't run into this again for the time being.

**Comment by shayne** [comment]
Thanks, @mvdan, I appreciate that, and I'm sorry for the issue you're experiencing. If you can come up with repro steps, that would help us track down the bug.

**Comment by mvdan** [comment]
Yep, for sure. Haven't run into the bug again for now, though I haven't been moving my laptop around as much the past couple of weeks. My working theory is still that tailscale somehow glitched after my machine temporarily lost its internet connection.

**Comment by DentonGentry** [comment]
Since there isn't enough information to make progress on, barring further comment we'll expect to close this during the next regular bug scrub.
